### PR TITLE
chore: Generate SLSA provenance for SBOM (#14438)

### LINKED ISSUE #14438: Use SLSA attestations to attest to SBOMs
Like any other artifacts, the SBOM can be attested to by SLSA attestations. (it's fairly common practice to attest to all built artifacts).

The project already attests to its binaries and container images via SLSA attestations, so should be easy to add the SBOMs.

I can send a PR updating the release workflow + documentation if you're interested.

### PR DESCRIPTION
**Pull request by enteraga6**
closes #14438 

This replaces Cosign with SLSA provenance generation and provides consistency with SLSA provenance generation for other artifacts. Verification is done with slsa-verifier; docs are updated to reflect this process.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
#14438 

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.

Tested release workflow on personal fork [here](https://github.com/enteraga6/argo-cd/releases/tag/v2.8.0-rc5) using this [workflow](https://github.com/enteraga6/argo-cd/blob/test/sbom-prov/.github/workflows/release.yaml)

SLSA Verifier Test took place on local CLI with successful result, verifying downloaded SBOM artifact and provenance:

```
Verified signature against tlog entry index 27409036 at URL: https://rekor.sigstore.dev/api/v1/log/entries/24296fb24b8ad77a45fee08f9a61155208816ebb8e5547295e6f07b12e3c5d82abab11df97a58083
Verified build using builder "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@refs/tags/v1.7.0" at commit ecace957a88916b139d1aba54eb0d031e1446981
Verifying artifact ../sbom.tar.gz: PASSED
```

* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -127,13 +127,15 @@ jobs:
...
 
   generate-sbom:
-    name: Create Sbom and sign assets
+    name: Create SBOM and generate hash
     needs:
       - argocd-image
       - goreleaser
     permissions:
       contents: write # Needed for release uploads
       id-token: write # Needed for signing Sbom
```
Comment (34fathombelow) [inline .github/workflows/release.yaml:136]: We can remove `id-token: write` permission as this will no longer be needed.

**Review by 34fathombelow — CHANGES_REQUESTED** [review]
Overall this is fantastic, thank you so much!   Just one minor change regarding the permissions change. Also if you don't mind changing the title of the PR to: `Chore: Generate SLSA provenance for SBOM (#14438) `

**Comment by 34fathombelow** [comment]
@crenshaw-dev Do we want to cherry-pick this to 2.8 and 2.7?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (enteraga6) [reply .github/workflows/release.yaml:136]: Great catch! Thank you! Changed title as well.

Comment (34fathombelow) [reply .github/workflows/release.yaml]: My bad, gave a horrible example of title description.    `chore` should be lowercase, sorry about that.

**Review by 34fathombelow — APPROVED** [review]
LGTM!  Thanks again!

File: .github/workflows/release.yaml (hunk 1 above)
Comment (enteraga6) [reply .github/workflows/release.yaml]: No worries!

**Review by crenshaw-dev — APPROVED** [review]

**Comment by crenshaw-dev** [comment]
@enteraga6 if you have time to open a cherry-pick PR against the release-2.8 branch, I'd love to have this as part of the 2.8 GA release. :-)

**Comment by enteraga6** [comment]
@crenshaw-dev Added the cherry-pick PR [here](https://github.com/argoproj/argo-cd/pull/14559)!

**Pull request merged**